inc/Enqueue.php

```diff
@@ -143,7 +143,7 @@ private function sanitize_path( $path ) {
 	 * and dynamic imports possible.
 	 */
 	public function printPublicPath() {
-		$publicPath = $this->getUrl( '' );
+		$publicPath = apply_filters( 'wpackio_print_public_path', $this->getUrl( '' ), $this->appName, $this->outputPath );
```

Comment:
While this could work, I think it would also create conflict when multiple plugins/themes are using Wpackio\Enqueue. Given that we are using `appName` to prevent conflict, I think a better approach would be

```php
$publicPath = apply_filters( 
  'wpackio_print_public_path_' . $this->appName, 
  $this->getUrl( '' ), 
  $this->appName, 
  $this->outputPath 
);
```

Then one can selectively filter it with `add_filter( 'wpackio_print_public_path_fooBarApp', ... )`. I hope I made sense, please let me know otherwise.

Reply:
I understand your concerns, that's why I added the `appName` as the second parameter, so it's possible to check for the correct app.

While your approach enforces this `appName` check and prevents potential incompatibility issues, it also prevents cases where I want to modify the `publicPath` for all plugins and themes at once.

e.g. when using a CDN
With my approach I could simply use this code to add CDN support:

```php
add_filter( 'wpackio_print_public_path', 'set_public_path_to_cdn' );

function set_public_path_to_cdn( $public_path ) {
	// wordpress get home url
	$home_url = get_home_url();
	$cdn_url = 'https://cdn.example.com';

	// replace wordpress home url with cdn url
	return str_replace($home_url, $cdn_url, $public_path);
}
```
This would change the URL to the CDN for all plugins and themes that use wpackio.

With your approach I had to add this filter for each plugin and theme manually.

But I understand your concerns. If third party plugins use wpackio, it may not always be clear, which plugins are affected with this filter.

Reply:
Hi @derweili, what you are saying is making complete sense. So please document it in the README.md file and we can go ahead and release it.

Reply:
@swashata I have updated the readme file as suggested.